packages/core/src/application.ts

```diff
@@ -133,6 +139,9 @@ export class Application extends Context {
   private _setupOperation(verb: string, path: string, spec: OperationObject) {
     const handler = spec['x-operation'];
     if (typeof handler === 'function') {
+      spec = Object.assign({}, spec);
```

Comment:
I am confused. `app.getApiSpec()` is performing a deep copy via `deep-assign`.  `getApiSpec` from `metadata.ts` is not declaring anything about shallow/deep copy.

Could you please clarify what do you mean and what changes would you like me to make?